ext/net/ops.rs

```diff
@@ -59,6 +63,40 @@ pub struct IpAddr {
   pub port: u16,
 }
 
+/// Processes and validates the hostname and port in the given `IpAddr` structure.
+///
+/// This function extracts the hostname and port from the provided `IpAddr` structure,
+/// validates them, and updates the structure with the correctly formatted hostname and port.
+/// This is necessary to ensure that the hostname and port are valid before they are used
+/// in network operations such as `Deno.connect` and `Deno.listen`.
```

Comment:
Do we need this function for this PR? It seems like a new feature, which I'm not sure we should do. Can we land this change in a separate PR?

Reply:
Yes, it is a new addition, however it is not complex or can be described as a new feature; it is simply an upgrade of existing behavior. 

**This is to extract the hostname from a URL and allow permission to access the hostname at a certain port, not the entire URL. For example, https://url:port/.


ity will offer permission for url at port from url rather than giving port at connect method. 


If we don't put:port, it will use the port argument in deno connect or deno listen.**

Reply:
I'd rather we not upgrade the existing behaviour for this bug fix and instead land that in a separate PR.

Reply:
I unresolved this comment—seems still relevant?

Reply:
This function is for the permission system because after I made changes to parsing the host and validating it to avoid the panic in allow-net, the permission system was affected, therefore I fixed it by introducing this function. and this not new feature

see this reference:
https://docs.deno.com/runtime/manual/basics/permissions
https://deno.land/api@v1.44.4?s=Deno.connect
https://deno.land/api@v1.44.4?s=Deno.listen